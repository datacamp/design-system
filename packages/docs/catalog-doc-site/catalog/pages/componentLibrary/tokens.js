import tokens from '@datacamp/waffles-tokens';
import { Page } from 'catalog';
import React from 'react';

import CustomHeader from '../../components/CustomHeader';

export default () => {
  return (
    <main>
      <CustomHeader
        section="Component Library"
        subSection="Tokens"
        description="The primitive values used to style anything and everything"
      />

      <Page>
        <section className="dc-u-maxw-100pc">
          <div className="dc-card dc-u-p-24 dc-u-mt-8">
            <h4>About</h4>
            <p>
              The <code>@datacamp/waffles-tokens</code> package exposes a json
              file containing all the primitive values necessary to produce apps
              with the datacamp styling rules. These values can be accessed and
              used to style applications using a variety of tools, including
              CSS-in-JS libraries such as emotion.
            </p>
            <h4>Usage</h4>
            <p>
              To use any of the styles, simply import the{' '}
              <code>@datacamp/waffles-tokens</code> package and directly access
              the necessary properties. For example if the primary color is
              needed then it should be accessed as follows.
            </p>
            <pre>
              <code>{`import { colors } from '@datacamp/waffles-tokens';

const primaryColorValue = colors.primary;`}</code>
            </pre>
            <p>
              This hex code can then be used wherever is necessary. All the
              tokens can be accessed and used in the same way since it is just a
              json file.
            </p>
            <h4>Available tokens</h4>
            <pre>
              <code>{JSON.stringify(tokens, null, 2)}</code>
            </pre>
          </div>
        </section>
      </Page>
    </main>
  );
};
