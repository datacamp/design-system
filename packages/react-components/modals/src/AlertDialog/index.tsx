import Button, { ButtonGroup } from '@datacamp/waffles-button';
import { Heading, Paragraph } from '@datacamp/waffles-text';
import { css } from '@emotion/core';
import React, { createRef } from 'react';

import BaseDialog, { CloseOrigin as BaseCloseOrigin } from '../BaseDialog';

interface AlertModalProps {
  cancelButtonText?: string;
  confirmButtonText?: string;
  dataAttributes?: { [key: string]: string };
  description: string;
  intent?: 'neutral' | 'success' | 'warning' | 'danger';
  isOpen: boolean;
  loading?: boolean;
  onClose: (origin: CloseOrigin) => void;
  onConfirm: () => void;
  title: string;
}
type CloseOrigin = 'cancelButton' | BaseCloseOrigin;

const AlertDialog: React.FC<AlertModalProps> = ({
  cancelButtonText = 'Cancel',
  confirmButtonText = 'Confirm',
  dataAttributes,
  description,
  intent = 'neutral',
  loading = false,
  isOpen,
  onClose,
  onConfirm,
  title,
}) => {
  // Callbacks to set the correct origin when closing
  const onCancelButton = (): void => onClose('cancelButton');

  // Focus on the button when the Modal opens
  const cancelButtonRef = createRef<HTMLButtonElement>();
  const confirmButtonRef = createRef<HTMLButtonElement>();
  const onAfterOpen = (): void => {
    const focusButtonRef =
      intent === 'neutral' || intent === 'success'
        ? confirmButtonRef
        : cancelButtonRef;
    if (focusButtonRef && focusButtonRef.current) {
      focusButtonRef.current.focus();
    }
  };

  return (
    <BaseDialog
      closeButtonDisabled={loading}
      contentLabel="Alert dialog"
      dataAttributes={dataAttributes}
      hideCloseButton={false}
      isOpen={isOpen}
      onAfterOpen={onAfterOpen}
      onClose={onClose}
      shouldCloseOnEsc={!loading}
      shouldCloseOnOverlayClick={!loading}
      width="480px"
    >
      <div css={css({ padding: 32, textAlign: 'center' })}>
        <Heading as="h1" size={600} multiLine>
          {title}
        </Heading>
        <Paragraph>{description}</Paragraph>
        <ButtonGroup css={{ marginTop: 24 }}>
          <Button
            ref={cancelButtonRef}
            disabled={loading}
            onClick={onCancelButton}
          >
            {cancelButtonText}
          </Button>
          <Button
            ref={confirmButtonRef}
            appearance="primary"
            intent={intent}
            loading={loading}
            onClick={onConfirm}
          >
            {confirmButtonText}
          </Button>
        </ButtonGroup>
      </div>
    </BaseDialog>
  );
};

export default AlertDialog;
