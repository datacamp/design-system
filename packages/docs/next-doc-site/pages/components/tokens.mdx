import Layout from '../../components/page-layout';
import Header from '../../components/page-header';
import Section from '../../components/page-section';
import FormattedTokens from '../../components/formatted-tokens';

export default Layout;

<Header
  pageTitle="Tokens"
  overview="The primitive values used to style anything and everything."
/>

<Section>

## Overview

The `@datacamp/waffles-tokens` package exposes all the primitive values necessary to style apps according to [Design Guidelines](https://www.figma.com/file/Flh00IKR70ff0bRvFW2Iou/%5BDC%5D-Design-Guidelines-Exploration?node-id=376%3A1144). These values can be accessed and used using a variety of tools, including CSS-in-JS libraries such as emotion.

### Usage

To use any of the styles, simply import the `@datacamp/waffles-tokens` package and directly access the necessary properties. For example if the green color is needed then it should be accessed as follows.

```
import tokens from '@datacamp/waffles-tokens';

const greenColor = tokens.colors.green;
```

This hex code can then be used wherever is necessary. All the tokens can be accessed and used in the same way.

### Colors

For each of the primary colours, there are 3 variations. The standard color e.g. `green` should be used as a priority, but there is also a dark and a light version of every colour. There is also a version to use as a text color when being used against a light coloured background, e.g. `greenDarkText`. For some colors this is the same as one of the other variants, but it will always have sufficient contrast against one of the lighter backgrounds. When working against a dark background, the standard colours are appropriate.

### Accessibility

Use `blue` (in some rare cases `blueDark` is allowed) color for interactive components focus styles. For disabled elements use `medium` opacity.

</Section>

<Section>

## Available Tokens

<FormattedTokens />

</Section>

<Section>

## Migration Guide

### Tokens (v1 -> v2)

Tokens have been updated to stay in line with [Design Guidelines](https://www.figma.com/file/Flh00IKR70ff0bRvFW2Iou/%5BDC%5D-Design-Guidelines-Exploration?node-id=376%3A1144) and to be compatible with [Figma Tokens](https://www.figma.com/community/plugin/843461159747178978/Figma-Tokens) plugin.

#### Deprecated tokens groups

- 游뚿 `spatial` group is deprecated
  - use tokens from `sizing`, `spacing`, and `zIndex` groups
- 游뚿 `border` group is deprecated
  - use tokens from `borderRadius` group
- 游뚿 `animation` group is deprecated

#### Completely new tokens groups

- 游릭 new group `sizing`
- 游릭 new group `spacing`
- 游릭 new group `borderWidth`
- 游릭 new group `opacity`
- 游릭 new group `zIndex`
- 游릭 new group `borderRadius`
  - 游뚿 `border.radius` -> `borderRadius.medium`
  - 游릭 new `circle` border radius token

#### Updated tokens groups

- 游댃 changes in `colors` group:
  - 游릭 new brand colors `brandFacebook`, `brandTwitter`, `brandGoogle`, `brandLinkedIn`
  - 游릭 new text colors `navySubtleTextOnDark`, `navySubtleTextOnLight`
  - 游릭 new `greyDark` color
  - 游뚿 `navyText` -> `navy`
  - 游뚿 `greenText` -> `greenDarkText`
  - 游뚿 `redText` -> `redDarkText`
  - 游뚿 `orangeText` -> `orangeDarkText`
  - 游뚿 `yellowText` -> `yellowDarkText`
  - 游뚿 `blueText` -> `blueDarkText`
  - 游뚿 `purpleText` -> `purpleDarkText`
  - 游뚿 `pinkText` -> `pinkDarkText`
  - 游뚿 `beige100` -> `beigeSubtle`
  - 游뚿 `beige200` -> `beigeLight`
  - 游뚿 `beige300` -> `beige`
  - 游뚿 `beige400` -> `beigeMedium`
  - 游뚿 `grey100` -> `greySubtle`
  - 游뚿 `grey200` -> `greyLight`
  - 游뚿 `grey300` -> `grey`
  - 游뚿 `grey400` -> `greyMedium`
- 游댃 changes in `breakpoints` group
  - 游릭 new `small`, `medium`, `large` breakpoints
  - 游뚿 all old tokens are deprecated
- 游댃 changes in `boxShadow` group
  - 游릭 new `thin`, `medium`, `thick`, `xthick` tokens
  - 游뚿 all old tokens are deprecated

#### Renamed and updated tokens groups

- 游댃 `fontFamily` -> `fontFamilies`
  - 游뚿 tokens now contains whole CSS font stacks, no need to specify additional fallback fonts
- 游댃 `fontSize` -> `fontSizes`
  - 游뚿 `micro` -> `small`
  - 游뚿 `small` -> `small`
  - 游뚿 `base` -> `medium`
  - 游뚿 `h6` -> `small`
  - 游뚿 `h5` -> `small`
  - 游뚿 `h4` -> `medium`
  - 游뚿 `h3` -> `large`
  - 游뚿 `h2` -> `xlarge`
  - 游뚿 `h1` -> `xxlarge`
  - 游릭 new `huge` token
- 游댃 `fontWeight` -> `fontWeights`
- 游댃 `letterSpacings` -> `letterSpacing`
  - 游뚿 `base` -> `default`
  - 游뚿 `mediumHeading` -> `tight`
  - 游뚿 `largeHeading` -> `tight`
  - 游릭 new `relaxed` token
- 游댃 `lineHeight` -> `lineHeights`
  - in most cases use the same `lineHeight` in tandem with `fontSize`
  - 游뚿 `base` -> `medium`
  - 游뚿 `heading` -> `large`
  - 游뚿 `largeHeading` -> `xlarge`
  - 游릭 new `small` token
  - 游릭 new `xxlarge` token

### Components

Most components feature only slight visual changes, without any updates to their props APIs (icons are the only exception).

#### Icons (v4 -> v5)

- 游뚿 `size` accepts `medium` (default), `small`, and `xsmall` values instead of numeric `24`, `18`, and `12`
  - in most cases, it is advised to use the default value
  - under the hood they map to 16, 14, and 12 numeric values

#### Button (v7 -> v8)

- font size for `small` and `medium` `Button` is set to `fontSizes.medium` (14px)
- adjusted icon size for `CloseButton` and social media buttons
- updated dropdown item elements dimensions for `UserAccountMenu`

#### Text (v5 -> v6)

- font size for `Small`, `Code`, and `CodeBlock` is set to `fontSizes.small` (12px)
- font size for `Paragraph`, `Link`, `Emphasis`, `Strong`, `Text`, and `ListItem` is set to `fontSizes.medium` (14px)
- font size, line height, and letter spacing are individually adjusted for each `Heading` size
- 丘멆잺 recommended combination of `size` for each HTML heading level for `Heading` component:
  - `h1` -> `700`
  - `h2` -> `650`
  - `h3` -> `600`
  - `h4` -> `500`
- font size for `small` `Badge` is set to `fontSizes.small` and for `large` one to `fontSizes.medium`

#### Toast (v3 -> v4)

- title font size is set to match the description
- position of icons are adjusted

#### Modals (v7 -> v8)

- reduce spacings around `Dialog.Header`, `Dialog.Body`, and `Dialog.Footer`

#### Tooltip (v0.2 -> v1)

- reduce spacings around the text

</Section>
