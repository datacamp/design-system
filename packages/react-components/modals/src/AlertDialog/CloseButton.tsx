import { CrossIcon } from '@datacamp/waffles-icons';
import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { css } from '@emotion/core';
import React from 'react';

const CloseButton: React.FC<{ onClick: () => void; disabled: boolean }> = ({
  onClick,
  disabled,
}) => (
  <button
    aria-label="Close"
    css={css({
      ':active': { color: tokens.color.opaque.primary.value.rgb },
      ':disabled': { color: '#D1D3D8' },
      ':disabled:hover, :disabled:focus': { backgroundColor: 'transparent' },
      ':focus': {
        backgroundColor: tokens.color.opaque.porcelain.value.rgb,
      },
      ':hover': { backgroundColor: tokens.color.opaque.porcelain.value.rgb },
      alignItems: 'center',
      backgroundColor: 'white',
      border: 'none',
      borderRadius: 2,
      color: tokens.color.opaque.greyDark.value.rgb,
      cursor: !disabled ? 'pointer' : 'not-allowed',
      display: 'flex',
      flexDirection: 'column',
      height: tokens.size.space[32].value,
      justifyContent: 'center',
      outline: 'none',
      padding: 0,
      position: 'absolute',
      right: tokens.size.space[8].value,
      top: tokens.size.space[8].value,
      width: tokens.size.space[32].value,
    })}
    disabled={disabled}
    onClick={!disabled ? onClick : undefined}
    type="button"
  >
    <CrossIcon size={18} />
  </button>
);

export default CloseButton;
