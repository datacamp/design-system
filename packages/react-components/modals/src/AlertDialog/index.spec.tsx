import '@testing-library/jest-dom/extend-expect';

import axeRender from '@datacamp/waffles-axe-render';
import tokens from '@datacamp/waffles-tokens/lib/future-tokens.json';
import { fireEvent, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import * as React from 'react';

import { setAppElement } from '..';
import AlertDialog from '.';

describe('<AlertModal />', () => {
  const baseContainer = document.createElement('div');
  setAppElement(baseContainer);
  const getRenderOptions = (): RenderOptions => ({
    baseElement: document.body,
    container: document.body.appendChild(baseContainer),
  });

  describe('isOpen', () => {
    it('renders the modal via a portal when true', async () => {
      const { container, getByText, baseElement, getByRole } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={() => {}}
          onConfirm={() => {}}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );

      expect(container).toHaveAttribute('aria-hidden', 'true'); // app gets hidden when open

      const modalElement = getByRole('dialog');
      expect(modalElement).toContainElement(getByText(
        'test title'
      ) as HTMLElement);
      expect(modalElement).toContainElement(getByText(
        'test description'
      ) as HTMLElement);
      expect(baseElement).toMatchSnapshot();
    });

    it('mounts nothing when false', async () => {
      const { container, baseElement, queryByRole } = await axeRender(
        <AlertDialog
          description="test description"
          isOpen={false}
          onClose={() => {}}
          onConfirm={() => {}}
          title="test title"
        />,
        getRenderOptions()
      );

      expect(container).not.toHaveAttribute('aria-hidden', 'true'); // app is interactive when hidden
      expect(queryByRole('dialog')).not.toBeInTheDocument();
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('confirmation', () => {
    it('uses a confirm button by default', async () => {
      const onConfirm = jest.fn();
      const { getByText } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={() => {}}
          onConfirm={onConfirm}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );
      const confirmationButton = getByText('Confirm') as HTMLElement;
      userEvent.click(confirmationButton);

      expect(onConfirm).toHaveBeenCalledTimes(1);
    });

    it('uses a custom confirm button', async () => {
      const onConfirm = jest.fn();
      const confirmButtonText = 'test confirm button';
      const { getByText } = await axeRender(
        <AlertDialog
          confirmButtonText={confirmButtonText}
          description="test description"
          onClose={() => {}}
          onConfirm={onConfirm}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );
      const confirmationButton = getByText(confirmButtonText) as HTMLElement;
      userEvent.click(confirmationButton);

      expect(onConfirm).toHaveBeenCalledTimes(1);
    });
  });

  describe('closing', () => {
    it('closes when clicking the X button', async () => {
      const onClose = jest.fn();

      const { getByTitle } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={onClose}
          onConfirm={() => {}}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );

      const closeButton = getByTitle('Cross') as HTMLElement;

      userEvent.click(closeButton);
      expect(onClose).toHaveBeenCalledTimes(1);
      expect(onClose).toHaveBeenCalledWith('closeButton');
    });

    it('closes when clicking the Cancel button with default text', async () => {
      const onClose = jest.fn();

      const { getByText } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={onClose}
          onConfirm={() => {}}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );

      const closeButton = getByText('Cancel') as HTMLElement;

      userEvent.click(closeButton);
      expect(onClose).toHaveBeenCalledTimes(1);
      expect(onClose).toHaveBeenCalledWith('cancelButton');
    });

    it('closes when clicking the Cancel button with custom text', async () => {
      const onClose = jest.fn();
      const cancelButtonText = 'cancel button text';
      const { getByText } = await axeRender(
        <AlertDialog
          cancelButtonText={cancelButtonText}
          description="test description"
          onClose={onClose}
          onConfirm={() => {}}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );

      const closeButton = getByText(cancelButtonText) as HTMLElement;

      userEvent.click(closeButton);
      expect(onClose).toHaveBeenCalledTimes(1);
      expect(onClose).toHaveBeenCalledWith('cancelButton');
    });

    it('closes when clicking the background overlay', async () => {
      const onClose = jest.fn();
      const { baseElement } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={onClose}
          onConfirm={() => {}}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );

      // need to use querySelector since there is no real content/label for the background
      const backgroundOverlayElement = baseElement.querySelector(
        '.modal-overlay'
      ) as HTMLElement;

      userEvent.click(backgroundOverlayElement);
      expect(onClose).toHaveBeenCalledTimes(1);
      expect(onClose).toHaveBeenCalledWith('overlayClick');
    });

    it('closes when clicking escape', async () => {
      const onClose = jest.fn();
      const { getByRole } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={onClose}
          onConfirm={() => {}}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );

      const dialogElement = getByRole('dialog') as HTMLElement;

      fireEvent.keyDown(dialogElement, {
        key: 'ESC',
        keyCode: 27,
      });

      expect(onClose).toHaveBeenCalledTimes(1);
      expect(onClose).toHaveBeenCalledWith('escKey');
    });
  });

  describe('loading', () => {
    it('renders a loading confirm button when loading', async () => {
      const { getByText, getByTitle } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={() => {}}
          onConfirm={() => {}}
          title="test title"
          isOpen
          loading
        />,
        getRenderOptions()
      );

      const confirmButtonElement = (getByText(
        'Confirm'
      ) as HTMLElement).closest('button');

      const spinnerElement = getByTitle('Spinner') as HTMLElement;
      expect(confirmButtonElement).toContainElement(spinnerElement);
    });

    it('disables the X button when loading', async () => {
      const onClose = jest.fn();

      const { getByTitle } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={onClose}
          onConfirm={() => {}}
          title="test title"
          isOpen
          loading
        />,
        getRenderOptions()
      );

      const closeButton = getByTitle('Cross') as HTMLElement;

      userEvent.click(closeButton);
      expect(onClose).not.toHaveBeenCalled();
    });

    it("doesn't close when clicking the Cancel button if loading", async () => {
      const onClose = jest.fn();

      const { getByText } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={onClose}
          onConfirm={() => {}}
          title="test title"
          isOpen
          loading
        />,
        getRenderOptions()
      );

      const closeButton = getByText('Cancel') as HTMLElement;

      userEvent.click(closeButton);
      expect(onClose).not.toHaveBeenCalled();
    });

    it("doesn't close when clicking the background overlay if loading", async () => {
      const onClose = jest.fn();
      const { baseElement } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={onClose}
          onConfirm={() => {}}
          title="test title"
          isOpen
          loading
        />,
        getRenderOptions()
      );

      // need to use querySelector since there is no real content/label for the background
      const backgroundOverlayElement = baseElement.querySelector(
        '.modal-overlay'
      ) as HTMLElement;

      userEvent.click(backgroundOverlayElement);
      expect(onClose).not.toHaveBeenCalled();
    });

    it("doesn't close when clicking escape if loading", async () => {
      const onClose = jest.fn();
      const { getByRole } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={onClose}
          onConfirm={() => {}}
          title="test title"
          isOpen
          loading
        />,
        getRenderOptions()
      );

      const dialogElement = getByRole('dialog') as HTMLElement;

      fireEvent.keyDown(dialogElement, {
        key: 'ESC',
        keyCode: 27,
      });

      expect(onClose).not.toHaveBeenCalled();
    });
  });

  describe('intent', () => {
    it('renders a neutral intent by default', async () => {
      const { getByText } = await axeRender(
        <AlertDialog
          description="test description"
          onClose={() => {}}
          onConfirm={() => {}}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );
      const confirmButtonElement = (getByText(
        'Confirm'
      ) as HTMLElement).closest('button');
      expect(confirmButtonElement).toHaveFocus();
      expect(confirmButtonElement).toHaveStyle(
        `background-color: ${tokens.color.opaque.primary.value.rgb}`
      );
    });

    it('renders a neutral intent', async () => {
      const { getByText } = await axeRender(
        <AlertDialog
          description="test description"
          intent="neutral"
          onClose={() => {}}
          onConfirm={() => {}}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );
      const confirmButtonElement = (getByText(
        'Confirm'
      ) as HTMLElement).closest('button');
      expect(confirmButtonElement).toHaveFocus();
      expect(confirmButtonElement).toHaveStyle(
        `background-color: ${tokens.color.opaque.primary.value.rgb}`
      );
    });

    it('renders a success intent', async () => {
      const { getByText } = await axeRender(
        <AlertDialog
          description="test description"
          intent="success"
          onClose={() => {}}
          onConfirm={() => {}}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );
      const confirmButtonElement = (getByText(
        'Confirm'
      ) as HTMLElement).closest('button');
      expect(confirmButtonElement).toHaveFocus();
      expect(confirmButtonElement).toHaveStyle(
        `background-color: ${tokens.color.opaque.green.value.rgb}`
      );
    });

    it('renders a warning intent', async () => {
      const { getByText } = await axeRender(
        <AlertDialog
          description="test description"
          intent="warning"
          onClose={() => {}}
          onConfirm={() => {}}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );
      const cancelButtonElement = (getByText('Cancel') as HTMLElement).closest(
        'button'
      );
      const confirmButtonElement = (getByText(
        'Confirm'
      ) as HTMLElement).closest('button');
      expect(cancelButtonElement).toHaveFocus();
      expect(confirmButtonElement).toHaveStyle(
        `background-color: ${tokens.color.opaque.orange.value.rgb}`
      );
    });

    it('renders a danger intent', async () => {
      const { getByText } = await axeRender(
        <AlertDialog
          description="test description"
          intent="danger"
          onClose={() => {}}
          onConfirm={() => {}}
          title="test title"
          isOpen
        />,
        getRenderOptions()
      );
      const cancelButtonElement = (getByText('Cancel') as HTMLElement).closest(
        'button'
      );
      const confirmButtonElement = (getByText(
        'Confirm'
      ) as HTMLElement).closest('button');
      expect(cancelButtonElement).toHaveFocus();
      expect(confirmButtonElement).toHaveStyle(
        `background-color: ${tokens.color.opaque.red.value.rgb}`
      );
    });
  });
});
