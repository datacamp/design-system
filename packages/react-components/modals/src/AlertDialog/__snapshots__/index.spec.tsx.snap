// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertModal /> isOpen mounts nothing when false 1`] = `
<body
  class=""
>
  <div
    class="ReactModalPortal"
  />
  <div />
  <div
    class="ReactModalPortal"
  />
</body>
`;

exports[`<AlertModal /> isOpen renders the modal via a portal when true 1`] = `
.emotion-11 {
  opacity: 1 !important;
  -webkit-transform: translate(-50%,-50%) scale(1) !important;
  -ms-transform: translate(-50%,-50%) scale(1) !important;
  transform: translate(-50%,-50%) scale(1) !important;
}

.emotion-10 {
  left: 50%;
  margin: auto;
  opacity: 0;
  outline: none;
  position: absolute;
  top: 50%;
  -webkit-transform: translate(-50%,-50%) scale(0.5);
  -ms-transform: translate(-50%,-50%) scale(0.5);
  transform: translate(-50%,-50%) scale(0.5);
  -webkit-transition: 0.4s cubic-bezier(0.19,1,0.22,1);
  transition: 0.4s cubic-bezier(0.19,1,0.22,1);
  width: 480px;
}

.emotion-13 {
  opacity: 1 !important;
}

.emotion-12 {
  background-color: rgba(38,62,99,0.8);
  bottom: 0;
  left: 0;
  opacity: 0;
  position: fixed;
  right: 0;
  top: 0;
  -webkit-transition: opacity 0.6s cubic-bezier(0.19,1,0.22,1);
  transition: opacity 0.6s cubic-bezier(0.19,1,0.22,1);
  z-index: 100;
}

.emotion-14 {
  overflow: hidden;
}

.emotion-0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: white;
  border: none;
  border-radius: 2px;
  color: rgb(61,66,81);
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 32px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  padding: 0;
  position: absolute;
  right: 8px;
  top: 8px;
  width: 32px;
  z-index: 1;
}

.emotion-0:active {
  color: rgb(51,170,204);
}

.emotion-0:disabled {
  color: #D1D3D8;
}

.emotion-0:disabled:hover,
.emotion-0:disabled:focus {
  background-color: transparent;
}

.emotion-0:focus {
  background-color: rgb(240,244,245);
}

.emotion-0:hover {
  background-color: rgb(240,244,245);
}

.emotion-9 {
  background-color: rgb(255,255,255);
  border-radius: 4px;
  box-shadow: 0px 0px 1px 0px rgba(61,66,81,0.3),0 4px 8px 0 rgba(0,0,0,0.15);
  -webkit-transition: all 0.6s cubic-bezier(0.075,0.82,0.165,1);
  transition: all 0.6s cubic-bezier(0.075,0.82,0.165,1);
}

.emotion-8 {
  padding: 32px;
  text-align: center;
}

.emotion-1 {
  -webkit-font-smoothing: antialiased;
  color: rgb(77,83,86);
  font-family: sans-serif;
  font-family: Lato;
  font-style: normal;
  color: rgb(61,66,81);
  font-weight: 700;
  line-height: 1.25;
  margin: 0;
  overflow: auto;
  white-space: normal;
  font-size: 24px;
  margin-top: 16px;
}

.emotion-1:first-child:not(style),
.emotion-1 style:first-child + * {
  margin-top: 0;
}

.emotion-2 {
  -webkit-font-smoothing: antialiased;
  color: rgb(77,83,86);
  font-family: sans-serif;
  font-family: Lato;
  font-style: normal;
  font-size: 16px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  margin-top: 8px;
}

.emotion-2:first-child:not(style),
.emotion-2 style:first-child + * {
  margin-top: 0;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  white-space: nowrap;
  margin-top: 24px;
}

.emotion-4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: 0.15s;
  transition: 0.15s;
  vertical-align: baseline;
  white-space: nowrap;
  background-color: transparent;
  border-color: #33aacc;
  color: #33aacc;
  padding: 0 15px;
  margin-left: 0;
}

.emotion-4:active {
  -webkit-transform: perspective(1px) scale(0.975);
  -ms-transform: perspective(1px) scale(0.975);
  transform: perspective(1px) scale(0.975);
}

.emotion-4:disabled,
.emotion-4:hover:disabled,
.emotion-4:active:disabled {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-4:focus {
  outline: 0;
}

.emotion-4:focus {
  box-shadow: 0 0 0 4px rgba(51,170,204,0.2);
}

.emotion-4:hover {
  border-color: #2E8FAB;
  color: #2E8FAB;
}

.emotion-3 {
  -webkit-font-smoothing: antialiased;
  color: rgb(77,83,86);
  font-family: sans-serif;
  font-family: Lato;
  font-style: normal;
  font-size: 16px;
  font-weight: 400;
  line-height: 16px;
  font-size: 16px;
  line-height: 46px;
  color: #3d4251;
  font-weight: bold;
}

.emotion-6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 4px;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: capitalize;
  -webkit-transition: 0.15s;
  transition: 0.15s;
  vertical-align: baseline;
  white-space: nowrap;
  background-color: transparent;
  border-color: #33aacc;
  color: #33aacc;
  background-color: #33aacc;
  padding: 0 15px;
  margin-left: 16px;
}

.emotion-6:active {
  -webkit-transform: perspective(1px) scale(0.975);
  -ms-transform: perspective(1px) scale(0.975);
  transform: perspective(1px) scale(0.975);
}

.emotion-6:disabled,
.emotion-6:hover:disabled,
.emotion-6:active:disabled {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.emotion-6:focus {
  outline: 0;
}

.emotion-6:focus {
  box-shadow: 0 0 0 4px rgba(51,170,204,0.2);
}

.emotion-6:hover {
  border-color: #2E8FAB;
  color: #2E8FAB;
}

.emotion-6:hover {
  background-color: #2E8FAB;
  border-color: #2E8FAB;
}

.emotion-5 {
  -webkit-font-smoothing: antialiased;
  color: rgb(77,83,86);
  font-family: sans-serif;
  font-family: Lato;
  font-style: normal;
  font-size: 16px;
  font-weight: 400;
  line-height: 16px;
  font-size: 16px;
  line-height: 46px;
  color: white;
  font-weight: bold;
}

<body
  class="emotion-14"
>
  <div
    aria-hidden="true"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="modal-overlay emotion-12 emotion-13"
    >
      <div
        aria-label="Alert dialog"
        class="emotion-10 emotion-11"
        role="dialog"
        tabindex="-1"
      >
        <button
          aria-label="Close"
          class="emotion-0"
          type="button"
        >
          <svg
            aria-hidden="false"
            height="18"
            role="img"
            viewBox="0 0 18 18"
            width="18"
          >
            <title>
              Cross
            </title>
            <path
              d="M9.005 7.625l4.83-4.83a.976.976 0 0 1 1.38 1.38l-4.83 4.83 4.82 4.82a.976.976 0 1 1-1.38 1.38l-4.82-4.82-4.83 4.83a.976.976 0 0 1-1.38-1.38l4.83-4.83-4.84-4.84a.976.976 0 1 1 1.38-1.38l4.84 4.84z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </button>
        <div
          class="emotion-9"
        >
          <div
            class="emotion-8"
          >
            <h1
              class="emotion-1"
            >
              test title
            </h1>
            <p
              class="emotion-2"
            >
              test description
            </p>
            <div
              class="emotion-7"
            >
              <button
                class="emotion-4"
                type="button"
              >
                <span
                  class="emotion-3"
                >
                  Cancel
                </span>
              </button>
              <button
                class="emotion-6"
                type="button"
              >
                <span
                  class="emotion-5"
                >
                  Confirm
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
